src/extensions/qdatabase.c

```diff
@@ -431,11 +536,41 @@ static qdbresult_t *execute_query(qdb_t *db, const char *query)
     result->cursor = 0;
 
     /* assign methods */
-    result->get_str = _resultGetStr;
-    result->get_str_at = _resultGetStrAt;
-    result->get_int = _resultGetInt;
-    result->get_int_at = _resultGetIntAt;
-    result->get_next = _resultGetNext;
+    result->get_str = result_get_str;
+    result->get_str_at = result_get_str_at;
+    result->get_int = result_get_int;
+    result->get_int_at = result_get_int_at;
+    result->get_next = result_get_next;
+
+    result->get_cols = result_get_cols;
+    result->get_rows = result_get_rows;
+    result->get_row = result_get_row;
+
+    result->free = result_free;
+
+    return result;
+#elif defined(Q_ENABLE_PGSQL)
+    // query
+    DEBUG("%s", query);
+    if (pgsql_query(db->pgsql, query, PQ_READ)) return NULL;
+
+    // store
+    qdbresult_t *result = (qdbresult_t *)malloc(sizeof(qdbresult_t));
+    if (result == NULL) return NULL;
+    result->pgsql = db->pgsql;
+
+    /* get meta data */
+    // result->row = NULL;
+    qtype_cast(pgsql_t*, result->pgsql)->rows = pgsql_num_rows(result->pgsql);
+    qtype_cast(pgsql_t*, result->pgsql)->cols = pgsql_num_fields(result->pgsql);
+    qtype_cast(pgsql_t*, result->pgsql)->cursor = 0;
+
+    /* assign methods */
+    result->get_str = result_get_str;
+    result->get_str_at = result_get_str_at;
+    result->get_int = result_get_int;
+    result->get_int_at = result_get_int_at;
+    result->get_next = result_get_next;
```

Comment:
looks like these are common and can ignore code duplication.
I'm not trying to remove 100% of code duplication but some level that isn't very hard to do and still provides good code readability. Can you review the change from that perspective once again?